src/_pytest/pathlib.py

```diff
@@ -541,16 +561,7 @@ def import_path(
         insert_missing_modules(sys.modules, module_name)
         return mod
 
-    pkg_path = resolve_package_path(path)
-    if pkg_path is not None:
-        pkg_root = pkg_path.parent
-        names = list(path.with_suffix("").relative_to(pkg_root).parts)
-        if names[-1] == "__init__":
-            names.pop()
-        module_name = ".".join(names)
-    else:
-        pkg_root = path.parent
-        module_name = path.stem
```

Comment:
Extracted to `resolve_pkg_root_and_module_name` so it can be reused inside the `importlib` mode.